src/Admin/AdminConsole/Controllers/ProvidersController.cs

```diff
@@ -177,14 +177,19 @@ public async Task<IActionResult> Edit(Guid id)
     [RequirePermission(Permission.Provider_Edit)]
     public async Task<IActionResult> Edit(Guid id, ProviderEditModel model)
     {
-        var providerPlans = await _providerPlanRepository.GetByProviderId(id);
         var provider = await _providerRepository.GetByIdAsync(id);
+
         if (provider == null)
         {
             return RedirectToAction("Index");
         }
 
-        model.ToProvider(provider);
+        provider.BillingEmail = model.BillingEmail?.ToLowerInvariant().Trim();
+        provider.BillingPhone = model.BillingPhone?.ToLowerInvariant().Trim();
+        provider.Gateway = model.Gateway;
+        provider.GatewayCustomerId = model.GatewayCustomerId;
+        provider.GatewaySubscriptionId = model.GatewaySubscriptionId;
```

Comment:
What's wrong with `model.ToProvider`? :(
(To expand slightly - I like putting this kind of mapping/update logic inside models, it makes sense to me from a general OOP perspective, and it removes boilerplate from the controller or service that uses it.)

Reply:
I personally just find the signature `public virtual Provider ToProvider(Provider existingProvider)` and the practice of passing around mutable object properties in general somewhat confusing, but I've reset it back to its original state and added the new Gateway fields to the existing method.

Reply:
That's reasonable - do you think it would be better if it was on the object being mutated instead, e.g. `existingProvider.Update(editProviderModel)`? That way the object is updating itself and not mutating its argument. (I suspect we've done it the current way to avoid importing api models into the core layer, but you could avoid this by making `Update` an api-only extension method on `Provider`.)

Anyway - none of that needs to be done here - just thinking aloud.

Reply:
Yeah I think that could be a bit more readable. Even just using a different signature for the same approach might make it easier to reason through at first glance as well. 

For instance, it makes perfect sense for `CreateProviderModel` to have a `ToProvider` method that returns a `Provider` entity that will be inserted into the DB. But for the `EditProviderModel`, we already have an existing `Provider`, so if we're going to update in place, there's no reason to return anything and the method name could be more along the lines of `EditProviderModel.UpdateEntity(Provider)` or even just `EditProviderModel.Update(Provider)`. If AC is open to that, I can definitely start making those changes as we make more Admin updates!

Reply:
I agree with `EditProviderModel.Update(Provider)`, returning `void`!